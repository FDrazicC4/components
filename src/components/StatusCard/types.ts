export type Variants = 'standard' | 'compact';
export type Colors = 'info' | 'success' | 'danger';

export interface Props {

    /**
     * Any optional additional class names
     */
    className?: string,

    /**
     * The variant of card you want to display
     */
    variant: Variants,

    /**
     * The colors available
     */
    color: Colors,

    /**
     * The header text to display
     */
    header: string,

    /**
     * The name of the Material Icon to display
     */
    icon: string,

    /**
     * The body text to display
     */
    body: string,

    /**
     * Optional disabled attribute to disable component
     */
    disabled?: boolean,

    /**
     * Optional fill attribute to fill background color
     */
    fill?: boolean,

}
