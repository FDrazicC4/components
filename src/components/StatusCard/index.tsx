import React from 'react';
import StandardStatusCard from './Components/Standard';
import CompactStatusCard from './Components/Compact';
import { Props } from './types';

const components = {
    standard: StandardStatusCard,
    compact: CompactStatusCard,
};

const StatusCard = ({ className, variant,color, header, body, disabled, fill, ...otherProps}: Props) => {

    const Element = components[variant] || CompactStatusCard;

    return <Element className={className} color={color} header={header} body={body} {...otherProps} />;
};

export default StatusCard;
