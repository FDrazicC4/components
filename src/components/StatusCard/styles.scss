@import '../../styles/_utilities.scss';

// Dashboard Cards are restricted to three colors (info, success, and danger)
// color(accent, info,success,danger)

.wrapper {
    box-sizing: border-box;
    box-shadow: 0 1px 3px rgba(0, 0, 0, 0.16);
    display: inline-block;
    margin: 1rem;
}
.standard {
    padding: 0.5rem 1rem;
    border-left: 0.25rem solid;
    margin: 1rem;
}

$accentColors: (info, success, danger);

@each $color in $accentColors {

    .color--#{$color} {
        border-left-color: color(accent, $color);

        &.disabled {
            border-left-color: color(brand,tertiary);

            & h1,
            span {
                color: color(brand,tertiary);
            }

            &.fill {
                background-color: rgba(color(brand, tertiary), .2);
            }
        }

        &.fill {
            background-color: rgba(color(accent, $color), .2);
        }

    }

}

