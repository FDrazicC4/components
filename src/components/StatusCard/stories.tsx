import React from 'react';
import { text, select, boolean } from '@storybook/addon-knobs';
import StatusCard from './index';

export default { title: 'Status Card'};

export const General = () => (

    <StatusCard
        header={text("Header", "hello")}
        body={text("Body", "goodbye")}
        color={select("Color", ["info", "success", "danger"], "info")}
        disabled={boolean("Disabled", false)}
        fill={boolean("Fill", false)}
        icon={text("Icon", "accessible")}
        variant={select("Variant", ["standard", "compact"], "standard")}
    />

);
