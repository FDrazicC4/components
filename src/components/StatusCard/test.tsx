import React from 'react';
import { shallow } from 'enzyme';
import StatusCard from './index';
import StandardStatusCard from './components/Standard';
import CompactStatusCard from './components/Compact';


describe('Status Card', () => {


    it('Renders Standard Variant', () => {
        const card = shallow(<StatusCard variant="standard" color="success" header="hi" body="bye" icon="business" />);
        expect(card.find(StandardStatusCard)).toHaveLength(1);
    });

    it('Renders Compact Variant', () => {
        const card = shallow(<StatusCard variant="compact" color="success" header="hi" body="bye" icon="" />);
        expect(card.find(CompactStatusCard)).toHaveLength(1);
    });


});


describe('Standard Status Card', () => {


    it('Adds the "brew-StatusCard--isDisabled" class when "isDisabled" prop exists', () => {
        expect(shallow(<StandardStatusCard body="de nada" color="success" header="gracias" icon="business" isDisabled />).hasClass('brew-StatusCard--isDisabled')).toBe(true);
    });

    it('Adds the "brew-StatusCard--isFilled" class when "isFilled" prop exists', () => {
        expect(shallow(<StandardStatusCard body="de nada" color="success" header="gracias" icon="business" isFilled />).hasClass('brew-StatusCard--isFilled')).toBe(true);
    });

    it('Adds the "brew-StatusCard__color--success" class when "danger" is passed to color prop', () => {
        expect(shallow(<StandardStatusCard body="de nada" color="success" header="gracias" icon="business" />).hasClass('brew-StatusCard__color--success')).toBe(true);
    });


});


describe('Compact Status Card', () => {


    it('Adds the "brew-StatusCard--isDisabled" class when "isDisabled" prop exists', () => {
        expect(shallow(<CompactStatusCard body="de nada" color="success" header="gracias" isDisabled />).hasClass('brew-StatusCard--isDisabled')).toBe(true);
    });

    it('Adds the "brew-StatusCard--isFilled" class when "isFilled" prop exists', () => {
        expect(shallow(<CompactStatusCard body="de nada" color="success" header="gracias" isFilled />).hasClass('brew-StatusCard--isFilled')).toBe(true);
    });

    it('Adds the "brew-StatusCard__color--danger" class when "danger" is passed to color prop', () => {
        expect(shallow(<CompactStatusCard body="de nada" color="danger" header="gracias" />).hasClass('brew-StatusCard__color--danger')).toBe(true);
    });


});