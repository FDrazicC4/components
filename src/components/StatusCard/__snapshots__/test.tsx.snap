// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Status Card Renders correctly 1`] = `
<div
  className="brew-StatusCard brew-StatusCard--standard brew-StatusCard--color-success"
>
  <section
    className="icon"
  >
    <span
      className="wrapper color--success size--inherit"
    >
      <svg
        height="100%"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        ‚òê
      </svg>
    </span>
  </section>
  <section>
    <h1
      className="brew-Typography brew-Typography--variant-h1 brew-Typography--color-success brew-Typography--align-inherit brew-Typography--display-initial"
    >
      hi
    </h1>
    <span
      className="brew-Typography brew-Typography--variant-body1 brew-Typography--color-success brew-Typography--align-inherit brew-Typography--display-initial"
    >
      bye
    </span>
  </section>
</div>
`;
